### PR TITLE
feat: add sticky review checklist on every PR

### PR DESCRIPTION
**Pull request by galargh**
The new workflow posts a checklist to PRs.

- posts only on PRs that are ready for review (no drafts)
- new commits reset the checklists
- checklist comment is sticky i.e. there's only one comment per PR (no duplicate comments)
- checklist item for verifying CI passed is always present
- checklist item for verifying CHANGELOG has been udpated is present only if Go files were modified (we could also check in the workflow if the relevant changelog was modified)

The checklist items are just examples and are meant to showcase what's possible with this setup. Ultimately, the team should decide what they'd find the most useful.

###### Testing

- this is how review checklist looks like in practice - https://github.com/galargh/kubo/pull/2#issuecomment-1398239428

### CODE REVIEW

**Comment by BigLep** [comment]
Cool @galargh !  I would share with the wider Kubo team to get suggestions on if there are other things that should add.

**Comment by Jorropo** [comment]
> checklist item for verifying CHANGELOG has been udpated is present only if Go files were modified (we could also check in the workflow if the relevant changelog was modified)

I think there is a more PRs that do not require CHANGELOG updates than ones that do.
Personally the only thing I care about here is seeing if CI runned and failed or not, which the github UI already do an awesome job as conveying.

**Comment by ajnavarro** [comment]
I like having a check to remember to add documentation or a new entry to the changelog, but I do not understand the CI checks. We already have these checks, right?

We can simplify this by adding these checks to our issue template, as a reminder of things we must do on every new PR.

**Comment by galargh** [comment]
With this PR I wanted to showcase how we could create sticky comments with reminders. As mentioned in the description, I think it's up to the team to decide what information should be delivered this way (if any). 

First, Let me remove the check `is CI green` because it's quite confusing. My intention was for it to only showcase checks that are always present as opposed to the ones that only appear under certain conditions and I didn't have a better idea at a time. 

Finally, here's an alternative for how we could communicate to contributors that a changelog update might be needed - https://github.com/ipfs/kubo/pull/9572.

Some differences worth considering that come to mind when it comes to sticky comments vs pull request templates:
- because they're created by workflows, we can customise sticky comments more
- pull request templates, unlike sticky comments, are visible to the contributors BEFORE a PR is created
- contributors can modify pull request template so the reviewers might not necessarily see it

File: .github/workflows/review-checklist.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            const versionRegex = /CurrentVersionNumber = "([^"]+)"/
+            const version = versionContent.match(versionRegex)[1]
+            const versionSplit = version.split(".")
+            const versionMajorMinor = `${versionSplit[0]}.${versionSplit[1]}`
+
+            const checks = []
+
+            // CI Checks
+            checks.push('Are all the CI checks 🟢?')
+
```
Comment (galargh) [inline .github/workflows/review-checklist.yml:45]:
```suggestion
```

Removing the confusing check.

**Comment by aschmahmann** [comment]
triage: @galargh is going to take a look this week

**Comment by galargh** [comment]
Closing in favor https://github.com/ipfs/kubo/pull/9620

**Pull request closed without merging**